sky/utils/subprocess_utils.py

```diff
@@ -92,35 +93,43 @@ def kill_children_processes(first_pid_to_kill: Optional[int] = None,
        etc. while we are cleaning up the clusters.
 
     Args:
-        first_pid_to_kill: Optional PID of a process to be killed first.
+        first_pid_to_kill: Optional PID of a process, or PIDs of a series of
+         processes to be killed first. If a list of PID is specified, it is
+         killed by the order in the list.
          This is for guaranteeing the order of cleaning up and suppress
          flaky errors.
     """
-    parent_process = psutil.Process()
+    first_to_kill_pid_to_processes = dict()
     child_processes = []
-    for child in parent_process.children(recursive=True):
-        if child.pid == first_pid_to_kill:
+    if isinstance(first_pid_to_kill, int):
+        first_pid_to_kill = [first_pid_to_kill]
+    elif first_pid_to_kill is None:
+        first_pid_to_kill = []
+
+    def _kill_processes(processes: List[psutil.Process]) -> None:
+        for process in processes:
             try:
                 if force:
-                    child.kill()
+                    process.kill()
                 else:
-                    child.terminate()
-                child.wait()
+                    process.terminate()
             except psutil.NoSuchProcess:
-                # The child process may have already been terminated.
+                # The process may have already been terminated.
                 pass
+
+    parent_process = psutil.Process()
+    for child in parent_process.children(recursive=True):
+        if child.pid in first_pid_to_kill:
+            first_to_kill_pid_to_processes[child.pid] = child
         else:
             child_processes.append(child)
 
-    for child in child_processes:
-        try:
-            if force:
-                child.kill()
-            else:
-                child.terminate()
-        except psutil.NoSuchProcess:
-            # The child process may have already been terminated.
-            pass
+    _kill_processes([
+        first_to_kill_pid_to_processes[proc]
+        for proc in first_pid_to_kill
+        if proc in first_to_kill_pid_to_processes
+    ])
```

Comment:
why not `_kill_process(list(first_to_kill_pid_to_processes.values()))`?

Reply:
Because we want to kill all process in `first_to_kill_pid_to_processes` with the same order as `first_pid_to_kill`, and the order of `dict.values()` is insert order, not the specified order.

Reply:
Good point! That makes sense.